### PR TITLE
🧑‍🏫 Expand documentation

### PR DESCRIPTION
**Pull request by jodator**
See #1474

I think I have covered most things, so additional updates will be provided as a follow-up.

In this PR I've made some changes to dev scripts, so see if you understand the naming change. Previously, those were not so obvious and now, IMO, they should match what we mock (node vs query-node).

As for the documentation, you can preview it on the [`docs`](https://github.com/Joystream/pioneer/tree/docs) branch.

### CODE REVIEW

**Comment by jodator** [comment]
ps.: Sorry for rebasing, but that's easier to update file renames than merging main :pensive:

File: packages/ui/README.md [hunk 1]
```diff
@@ -1,30 +1,56 @@
...
+    * `forum` - [forum subsystem](https://joystream.gitbook.io/joystream-handbook/subsystems/forum)
+    * `memberships` - [memberships management](https://joystream.gitbook.io/joystream-handbook/subsystems/membership)
+    * `working-groups` – [working group governance](https://joystream.gitbook.io/joystream-handbook/governance/working-groups)
+    * `proposals` – [proposal system](https://joystream.gitbook.io/joystream-handbook/governance/proposals)
+* `app` – Anything related to the application
+  * assets
+  * routing
+  * pages
+  * global providers
+* `mocks` Application specific: routing, pages, etc.
```
Comment (DevNikita16) [inline packages/ui/README.md:20]: I don't think that we store **application specifics** in `mocks`. We just store **raw mocks** and **seeds** there

**Review by DevNikita16 — COMMENTED** [review]
@jodator 👏
I find the new namings easily understandable. Docs also looks nice!

File: packages/ui/README.md (hunk 1 above)
Comment (jodator) [reply packages/ui/README.md:20]: I've copied and pasted that and forgot to change it. Good catch, thanks.

**Review by p-sad — APPROVED** [review]
I like it. I only have some minor suggestions on typos/style

**Comment by jodator** [comment]
@thesan I'm merging this PR, but feel free to review it later and give your feedback as well :)

**Review by thesan — COMMENTED** [review]
@jodator sorry I was late on this.

The documentation is really good :heart_eyes:, thanks!

I just had a few syntax changes suggestions (in places I would be slightly confused while reading), and more importantly all the:
```
[...](packages/...)
```
links are broken.

Should we fix this in another branch ?

**Pull request merged**